### PR TITLE
feat: add change room name, issue name; modified data flow via socket

### PR DESCRIPTION
**Pull request by TMNhat1810**
-add feature change room name, issue name
-add fullConsensus check on reveal

### CODE REVIEW

File: app/utils/history.js [hunk 1]
```diff
@@ -20,9 +20,16 @@ export const getVoteSummary = (votes) => {
...
     results = coffeeTime ? EXTRA_VOTE.COFFEE : EXTRA_VOTE.QUESTION_MARK;
+    if (results === EXTRA_VOTE.COFFEE) fullConsensus = false;
   } else {
     results = `${voteSum / numbericVoteCount}`;
+    for (let i = 0; i < votes.length; i += 1) {
+      if (votes[i].vote !== results || votes[i].vote === EXTRA_VOTE.COFFEE) {
+        fullConsensus = false;
+        break;
+      }
+    }
```
Comment (ces-anduynguyen) [inline app/utils/history.js:31]: Why do we need this? Are we looping `votes` again inside `votes.forEach()`?

File: app/utils/history.js [hunk 2]
```diff
@@ -20,9 +20,19 @@ export const getVoteSummary = (votes) => {
...
     results = `${voteSum / numbericVoteCount}`;
+    for (let i = 0; i < votes.length; i += 1) {
+      if (
+        votes[i].vote &&
+        (votes[i].vote !== results || votes[i].vote === EXTRA_VOTE.COFFEE)
+      ) {
+        fullConsensus = false;
+        break;
+      }
+    }
```
Comment (ces-anduynguyen) [inline app/utils/history.js:34]: should extract this into another function that returns `fullConsensus` status
`fullConsensus = getFullConsensus(votes)`

File: app/utils/history.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { EXTRA_VOTE } from '../../constants/db.constants';
 export const getVoteSummary = (votes) => {
   let [results, voteOnTotal, playerResults] = ['', '', ''];
   let [voteCount, numbericVoteCount, voteSum] = [0, 0, 0];
-  let coffeeTime = false;
+  let [coffeeTime, fullConsensus] = [false, true];
 
   votes.forEach((userVoting) => {
     if (userVoting.vote) {
```
Comment (ces-anduynguyen) [inline app/utils/history.js:9]:
```
votes.forEach((userVoting) => {
    const { vote, username } = userVoting;
    if (vote) {
      voteCount += 1;
      playerResults += `${username} (${vote}), `;
      if (vote === EXTRA_VOTE.COFFEE) coffeeTime = true;
      else if (vote !== EXTRA_VOTE.QUESTION_MARK) {
        numbericVoteCount += 1;
        voteSum += parseInt(userVoting.vote);
      }
    }
```

**Review by ces-anduynguyen — CHANGES_REQUESTED** [review]

File: app/utils/history.js (hunk 3 above)
Comment (TMNhat1810) [reply app/utils/history.js:9]: got it

File: app/utils/history.js (hunk 2 above)
Comment (TMNhat1810) [reply app/utils/history.js:34]: got it

**Review by ces-anduynguyen — APPROVED** [review]

**Pull request merged**